test/server.js

```diff
@@ -72,20 +72,31 @@ var smokeTestsInterval = setInterval(() => {
       // Not sure if we want to stop testing after an error occurs
       // clearInterval(smokeTestsInterval);
     }
-    logger.synclog(stdout);
+    logger.sync.log(stdout);
   });
 }, env.test_interval);
 
-// server is given to https as a callback after it sets up https
-https
-  .createServer({
-    key: fs.readFileSync(env.tls.keyPath),
-    cert: fs.readFileSync(env.tls.certPath),
-    // passphrase: 'ninnies'
-  }, server)
-  .listen(METRICS_SERVER_PORT, () => {
-    logger.sync.log(`https listening on ${METRICS_SERVER_PORT}`);
-  });
+
+var pollForCert = setTimeout(() => {
```

Comment:
I don't think this should be necessary if you're using certs from a secret... The pod won't start until the secret it needs exists.

Reply:
That's the way I thought it worked as well, and didn't mess with this a couple days ago when it started failing.  I assume the `mount` was a race, that I used to win, but now am not certain.

Reply:
Is it failing using `docker run` or running in OpenShift?

Reply:
In `openshift`, specifically when I `oc process -f <template> | oc create -f -`.  I believe the mount is created, but the secrete `data` key & cert are populated asyncronously, so its a race condition (must be a race against a watch?).  

It wasn't a problem when I was manually creating the objects separately with the `yaml` in the `/kube` dir.